derive/src/lib.rs

```diff
@@ -75,6 +75,15 @@ fn wrap_with_dummy_const(input: DeriveInput, impl_block: proc_macro2::TokenStrea
 
 /// Derive `parity_scale_codec::Encode` and `parity_scale_codec::EncodeLike` for struct and enum.
 ///
+/// # Top level attributes
+///
+/// By default the macro will try to bound the types needed to implement their trait, but the
+/// bounds can be specified manually with the top level attributes:
+/// * `#[codec(encode_bound(T: Encode))]`: a custom where bound that will be used when deriving the
+///   `Encode` trait.
+/// * `#[codec(decode_bound(T: Decode))]`: a custom where bound that will be used when deriving the
+///   `Decode` trait.
```

Comment:
```suggestion
/// By default the macro will add [`Encode`] and [`Decode`] bounds to all types, but the
/// bounds can be specified manually with the top level attributes:
/// * `#[codec(encode_bound(T: Encode))]`: a custom bound added to the `where`-clause when deriving the
///   `Encode` trait, overriding the default.
/// * `#[codec(decode_bound(T: Decode))]`: a custom bound added to the `where`-clause when deriving the
///   `Decode` trait, overriding the default.
```

maybe?

Reply:
yes thanks, in both case what we mean by `all types` is a bit vague but I think it is fine.